src/ess/loki/workflow.py

```diff
@@ -1,24 +1,139 @@
+
+
+def _empty_iofq(_: ScatteringRunType) -> IofQ[ScatteringRunType]:
+    return IofQ[ScatteringRunType](
+        sc.DataArray(
+            data=sc.zeros(dims=["Q"], shape=[100]),
             coords={
-                'Q': sc.linspace(
-                    dim='Q', start=0.01, stop=0.3, num=101, unit='1/angstrom'
+                "Q": sc.linspace(
+                    dim="Q", start=0.01, stop=0.3, num=101, unit="1/angstrom"
                 )
             },
         )
-    }
+    )
+
+
+def _hist_iofq(i_of_q: IofQ[ScatteringRunType]) -> IofQHistogram[ScatteringRunType]:
+    return IofQHistogram[ScatteringRunType](i_of_q.hist("Q"))
+
+
+class LiveWorkflow:
+    """LoKi workflow for live data reduction."""
+
+    def __init__(self) -> None:
+        self.pipeline = self._build_pipeline()
+
+    def _build_pipeline(self) -> sciline.Pipeline:
+        """Build a workflow pipeline for live data reduction.
+
+        Returns
+        -------
+        :
+            A pipeline for live data reduction.
+            The initial pipeline will be missing the input data.
+            It should be set before calling the pipeline.
+
+        """
+        # Wavelength binning parameters
+        wavelength_min = sc.scalar(1.0, unit="angstrom")
+        wavelength_max = sc.scalar(13.0, unit="angstrom")
+        n_wavelength_bins = 50
+
+        providers = (
+            load_nexus_monitor,
+            load_nexus_source,
+            get_source_position,
+            get_monitor_data,
+            patch_monitor_data,
+            monitor_to_tof,
+            sans_monitor,
+            monitor_to_wavelength,
+            _hist_monitor_wavelength,
+            _empty_iofq,
+            _hist_iofq,
+        )
+
+        params = {
+            NeXusMonitorName[Incident]: "monitor_1",
+            NeXusMonitorName[Transmission]: "monitor_2",
+            WavelengthBins: sc.linspace(
+                "wavelength", wavelength_min, wavelength_max, n_wavelength_bins + 1
+            ),
+        }
+        workflow = sciline.Pipeline(providers, params=params)
+        return workflow
+
+    def __call__(self, group: JSONGroup) -> dict[str, sc.DataArray]:
+        """
+
+        Returns
+        -------
+        :
+            Plottable Outputs:
+
+            - MonitorHistogram[SampleRun, Incident]
+            - MonitorHistogram[SampleRun, Transmission]
+            - IofQHistogram[SampleRun]
+
+        """
+        # ``JsonGroup`` is turned into the ``NexusGroup`` here, not in the ``beamlime``
+        # so that the workflow can control the definition of the group.
+        self.pipeline[FilePath[Filename[SampleRun]]] = snx.Group(
+            group, definitions=snx.base_definitions()
+        )
+        results = self.pipeline.compute(
+            (
+                MonitorHistogram[SampleRun, Incident],
+                MonitorHistogram[SampleRun, Transmission],
+            )
+        )
+
+        return {str(tp): result for tp, result in results.items()}
```

Comment:
Missing `IofQ`? You could keep that workflow separate, and add this one as a second plugin.